src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -478,6 +478,13 @@ private void CheckForHttp11ConnectionInjection()
                 return;
             }
 
+            if (NetEventSource.Log.IsEnabled())
```

Comment:
Would it be possible to add some telemetry counters? So that you wouldn't need to turn on the whole debugging logs to diagnose it.

Reply:
What kind of counters did you have in mind?
We could expose many things, but they may not be too useful in diagnosing an issue like this since they are all aggregated for the process.

I see `System.Net.Sockets` doesn't have a "current" connection attempt counter - I don't see a reason not to add that. We have "current" counters on all the other sources.
Something like `PendingHttp11ConnectionCount` on Http would then be sort of redundant.

Reply:
What I was trying to point out is that in order to get this diagnostics log, you need to turn on the internal logging. It's very verbose and chaotic and not designed to be used in production. I see it mostly as our internal logging helping us diagnose issues. In general, we don't recommend customers to turn this on. And there were even attempts to get completely rid of this.
So the question is whether this trace should help us or the customer diagnose the issue? If it's for customers, shouldn't it be part of the official HTTP telemetry?

Reply:
EventSource does have the LogLevel for a reason, so we could certainly expose a lot more info under debug/trace.

The "official" telemetry surface is an API we effectively promise not to break.
As such, these logs shouldn't be unique events with a bunch of parameters since they are likely to change/break in the future.

We can, however, have something like
```c#
ConnectionPoolDebug(long poolId, string message);
```

Where you can still easily collect and aggregate all the information around the pool behavior, it just wouldn't be ideal for automatic processing.

Alternatively, we can start using the Keywords feature of EventSource (filtering by topic). Or the connection pool can have its own EventSource `System.Net.Http.ConnectionPool`.